### DISCUSSION TITLE
モノレポ化の相談

### DISCUSSION THREAD

**Discussion by nekono-dev** [nekono-dev/skyshare#105] [Ideas]
これまではCloudflare Pagesによるビルドに依存していたため、モノレポを使うことができませんでした。
参考: https://developers.cloudflare.com/pages/platform/known-issues/

しかし現在本レポジトリはGithub Actionによって動作しており、CloudflareへのデプロイもGithub Actionsでwrangler deployしているだけなので、やろうと思えば可能だと思っています。

ここで、本リポジトリをモノレポ化するか、バックエンドとフロントエンドをバラけさせたほうがいいのか、はたまたこのままにしちゃうかという部分で悩んでおります。
ご意見伺いたいです。 @ZEKE320 @so-asano @takecchi （コントリビューター/コラボレータ以外でも @everyone ）

**Comment by takecchi** [comment]
おぉ。であればバックエンドとフロントエンドで分けて管理するよりモノレポのほうが良いと思います！

**Comment by ZEKE320** [comment]
@nkte8 
恐らくモノレポ化するというのは、「`node_modules/*`や`.eslint.rc`などを共通化すること」と思っているのですが、具体的なモノレポ化の方針について確認しても良いでしょうか？それぞれメリット・デメリットありそうですね

**Reply by nekono-dev** [reply]
確認ありがとうございます！
@nkte8 的には単に`root`ディレクトリから各パッケージを管理できるようになること(`node_modules/*`の共通化？)だと思っていました。なのでlinterやformatterはそれぞれの設定を持ち越せると思っていました。
モノレポになると`.eslint.rc`なども共通化しないとメリットは薄い感じですかね...？（いずれはフロントエンド・バックエンドいずれのコードも、誰が見ても見やすいコードにしたい/するためにlinter/formatterの導入はしたいと思っています）

ちょっと具体的なイメージがついていなかった、モノレポ自体あまり経験がなかったので以下みたいな記事や、可読性があがるのかなとやんわりにしか考えられていませんでした...
https://zenn.dev/anneau/articles/4c9beff9645af7

@takecchi よければモノレポについてどういう部分でプロジェクトにメリットがあるのか（メリットを生み出せるのか）よければ教えていただきたいです！
